### PR TITLE
docs: Add Layout documentation

### PR DESCRIPTION
**Pull request by kraimbaud**
### ❓ Type of change

- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving doc)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Add usage of template and slots in documentation as is was complicated to find

### CODE REVIEW

File: content/docs/components/layouts.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+summary: Learn how to use layouts using slots
+---
+
+# Layouts
+
+You can use slots to render templates. Let's create a `index.edge` file inside the `views/components/layout` directory.
```
Comment (RomainLanz) [inline content/docs/components/layouts.md:7]:
```suggestion
You can use slots to render templates. Let's create a `app.edge` file inside the `views/components/layout` directory.
```

File: content/docs/components/layouts.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+```edge
+<!DOCTYPE html>
```
Comment (RomainLanz) [inline content/docs/components/layouts.md:10]:
```suggestion
// title: views/components/layout/app.edge
<!DOCTYPE html>
```

File: content/docs/components/layouts.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  {{{ await $slots.main() }}}
+</body>
+</html>
+```
+
+Edit a `welcome.edge` file inside the `views` directory.  
+In this exemple we want to render the main content of or welcome page using our `$slots.main` function, add a meta markup with `$slots.meta`, and change the default title. 
+
+```edge
+@layout({ title: "Welcome page title" })
```
Comment (RomainLanz) [inline content/docs/components/layouts.md:29]:
```suggestion
```edge
// title: views/welcome.edge
@layout.app({ title: "Welcome page title" })
```

**Review by RomainLanz — CHANGES_REQUESTED** [review]

**Comment by RomainLanz** [comment]
Hey! 👋🏻 

Thanks for your PR!

I am not sure if this section should be inside the `Component` category or in the `Guides` category.
What are your thoughts @thetutlage @Julien-R44?

**Comment by kraimbaud** [comment]
@RomainLanz I have made your change ;)

**Comment by RomainLanz** [comment]
Thanks!

**Pull request merged**